sdk/testing/stepwise/stepwise_test.go

```diff
@@ -0,0 +1,336 @@
+	l      sync.Mutex
+
+	teardownCalled bool
+}
+
+// Setup creates the mock environment, establishing a test HTTP server
+func (m *mockEnvironment) Setup() error {
+	mux := http.NewServeMux()
+	// LIST
+	mux.HandleFunc("/v1/test/keys", func(w http.ResponseWriter, req *http.Request) {
+		listResp := api.Secret{
+			RequestID: "list-request",
+			LeaseID:   "lease-id",
+		}
+		out, err := jsonutil.EncodeJSON(listResp)
+		if err != nil {
+			panic(err)
+		}
+		w.Write(out)
+	})
+	// lease revoke
+	mux.HandleFunc("/v1/sys/leases/revoke", func(w http.ResponseWriter, req *http.Request) {
+		w.Write(nil)
+	})
+	// READ
+	mux.HandleFunc("/v1/test/keys/name", func(w http.ResponseWriter, req *http.Request) {
+		readResp := api.Secret{
+			RequestID: "read-request",
+			LeaseID:   "lease-id",
+		}
+		out, err := jsonutil.EncodeJSON(readResp)
+		if err != nil {
+			panic(err)
+		}
+		w.Write(out)
+	})
+	// fall through that rejects any other url than "/"
+	mux.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
+		if req.URL.Path != "/" {
+			http.NotFound(w, req)
+			return
+		}
+		fmt.Fprintf(w, "{}")
+	})
+	m.ts = httptest.NewServer(mux)
+
+	return nil
+}
+
+// Client creates a Vault API client configured to the mock environment's test
+// server
+func (m *mockEnvironment) Client() (*api.Client, error) {
+	m.l.Lock()
+	defer m.l.Unlock()
+	if m.ts == nil {
+		return nil, errors.New("client() called but test server is nil")
+	}
+
+	if m.client != nil {
+		return m.client, nil
+	}
+
+	cfg := api.Config{
+		Address:    m.ts.URL,
+		HttpClient: cleanhttp.DefaultPooledClient(),
+		Timeout:    time.Second * 5,
+		MaxRetries: 2,
+	}
+
+	client, err := api.NewClient(&cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	m.client = client
+	return client, nil
+}
+
+func (m *mockEnvironment) Teardown() error {
+	m.teardownCalled = true
+	m.ts.Close()
+	return nil
+}
+
+func (m *mockEnvironment) Name() string { return "" }
+
+func (m *mockEnvironment) MountPath() string {
+	return "/test/"
+}
+
+func (m *mockEnvironment) RootToken() string { return "" }
+
+func stepFunc(path string, operation Operation, shouldError bool) Step {
+	s := Step{
+		Operation: operation,
+		Path:      path,
+	}
+	if shouldError {
+		s.Assert = func(resp *api.Secret, err error) error {
+			return errors.New("some error")
+		}
+	}
+	return s
+}
+
+// mockT implements TestT for testing
+type mockT struct {
+	ErrorCalled bool
+	ErrorArgs   []interface{}
+	FatalCalled bool
+	FatalArgs   []interface{}
+	SkipCalled  bool
+	SkipArgs    []interface{}
+
+	f bool
+}
+
+func (t *mockT) Error(args ...interface{}) {
+	t.ErrorCalled = true
+	t.ErrorArgs = args
+	t.f = true
+}
+
+func (t *mockT) Fatal(args ...interface{}) {
+	t.FatalCalled = true
+	t.FatalArgs = args
+	t.f = true
+}
+
+func (t *mockT) Skip(args ...interface{}) {
+	t.SkipCalled = true
+	t.SkipArgs = args
+	t.f = true
+}
+
+func (t *mockT) failed() bool {
+	return t.f
+}
+
+func (t *mockT) failMessage() string {
```

Comment:
This (and `failed()`) no longer seem to be referenced elsewhere.

Reply:
Those were copied over from the legacy framework, thanks for catching that. I removed them in 1bb36b0ab